x/protocolpool/keeper/keeper.go

```diff
@@ -244,20 +233,25 @@ func (k Keeper) IterateAndUpdateFundsDistribution(ctx context.Context) error {
 
 			_, ok := toDistribute[f.Recipient]
 			if !ok {
-				toDistribute[f.Recipient] = math.ZeroInt()
+				toDistribute[f.Recipient] = sdk.NewCoins()
+			}
+
+			for _, denom := range amount.Amount.Denoms() {
+				am := sdk.NewCoin(denom, f.Percentage.MulInt(amount.Amount.AmountOf(denom)).TruncateInt())
+				toDistribute[f.Recipient] = toDistribute[f.Recipient].Add(am)
+				fullAmountToDistribute = fullAmountToDistribute.Add(am)
 			}
-			amountToDistribute := f.Percentage.MulInt(amount).TruncateInt()
-			toDistribute[f.Recipient] = toDistribute[f.Recipient].Add(amountToDistribute)
-			fullAmountToDistribute = fullAmountToDistribute.Add(amountToDistribute)
 		}
 
 		// sanity check for max percentage
 		if percentageToDistribute.GT(math.LegacyOneDec()) {
 			return true, errors.New("total funds percentage cannot exceed 100")
 		}
 
-		remaining := math.LegacyOneDec().Sub(percentageToDistribute).MulInt(amount).RoundInt()
-		poolFunds = poolFunds.Add(remaining)
+		for _, denom := range amount.Amount.Denoms() {
+			remaining := sdk.NewCoin(denom, math.LegacyOneDec().Sub(percentageToDistribute).MulInt(amount.Amount.AmountOf(denom)).TruncateInt())
```

Comment:
Similar code was used before but is another round down which has a rest that is not distributed.
Better remaining = amount - distributed amount

Reply:
should be fixed now